# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CanvasChild.h
# Commit: 57fc38773fe3
# Full Hash: 57fc38773fe37f21748092e3e58fdf5265e85c0d
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2023-08-30 04:26:43
# Regressor Bug: 1841891
# File Overlap Count: 3
# Description:
#   Bug 1841891: Only use a larger canvas ring buffer when in the foreground. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D186544
# ==============================================================================

diff -r 082442e3b925 -r 57fc38773fe3 gfx/layers/ipc/CanvasChild.h
--- a/gfx/layers/ipc/CanvasChild.h	Tue Aug 29 19:15:34 2023 +0000
+++ b/gfx/layers/ipc/CanvasChild.h	Tue Aug 29 19:17:45 2023 +0000
@@ -35,6 +35,11 @@
    */
   static bool Deactivated() { return mDeactivated; }
 
+  /**
+   * Release resources until they are next required.
+   */
+  static void ClearCachedResources();
+
   ipc::IPCResult RecvNotifyDeviceChanged();
 
   ipc::IPCResult RecvDeactivate();
@@ -136,12 +141,12 @@
   static const uint32_t kCacheDataSurfaceThreshold = 10;
 
   static bool mDeactivated;
+  static bool mInForeground;
 
   RefPtr<CanvasDrawEventRecorder> mRecorder;
   TextureType mTextureType = TextureType::Unknown;
   uint32_t mLastWriteLockCheckpoint = 0;
   uint32_t mTransactionsSinceGetDataSurface = kCacheDataSurfaceThreshold;
-  TimeStamp mLastNonEmptyTransaction = TimeStamp::NowLoRes();
   std::vector<RefPtr<gfx::SourceSurface>> mLastTransactionExternalSurfaces;
   bool mIsInTransaction = false;
   bool mHasOutstandingWriteLock = false;